docs/source/future/core/nrunner.rst

```diff
@@ -53,54 +53,166 @@ Runnable
+network URI, such as ``https://example.org:8080``.
+
+Runnable examples
++++++++++++++++++
+
+Possibly the simplest example for the use of a Runnable is to describe
+how to run a standalone executable, such as the ones available on your
+``/bin`` directory.
+
+As stated earlier, a runnable must declare its kind.  For standalone
+executables, a name such as ``exec`` fulfills the naming suggestions
+given earlier.
+
+A Runnable can be created in a number of ways.  The first one is
+through :class:`avocado.core.nrunner.Runnable`, a very low level (and
+internal) API.  Still, it serves as an example::
+
+  >>> from avocado.core import nrunner
+  >>> runnable = nrunner.Runnable('exec', '/bin/true')
+  >>> runnable
+  <Runnable kind="exec" uri="/bin/true" args="()" kwargs="{}" tags="None">
+
+The second way is through a JSON based file, which, for the lack of a
+better term, we're calling a (Runnable) "recipe".  The recipe file
+itself will look like::
+
+  {"kind": "exec", "uri": "/bin/true"}
+
+And example the code to create it::
+
+  >>> from avocado.core import nrunner
+  >>> runnable = nrunner.Runnable.from_recipe("/path/to/recipe.json")
+  >>> runnable
+  >>> <Runnable kind="exec" uri="/bin/true" args="()" kwargs="{}" tags="None">
+
+The third way to create a Runnable, is even more internal.  Its usage
+is **discouraged**, unless you are creating a tool that needs to
+create Runnables based on the user's input from the command line::
+
+  >>> from avocado.core import nrunner
+  >>> runnable = nrunner.Runnable.from_args({kind: 'exec', uri: '/bin/true'})
+  >>> runnable
+  >>> <Runnable kind="exec" uri="/bin/true" args="()" kwargs="{}" tags="None">
 
 Runner
 ~~~~~~
 
-A runner is an active entity that acts on the information of a
-runnable.  A runner will usually be tied to an specific kind of
-runnable, and will to able to act upon the specific information that
-runnable kind provides.
+A Runner, within the context of the N(ext)Runner architecture, is an
+active entity that acts on the information that a runnable contains,
+and quite simply, should be able to run what the Runnable describes.
+
+A Runner will usually be tied to a specific ``kind`` of Runnable.
+That type of relationship (Runner is capable of running kind "foo"
+and Runnable is of the same kind "foo") is the expected mechanism that
+will be employed when selecting a Runner.
+
+A Runner can take different forms, depending on which layer one is
+interacting with.  At the lowest layer, a Runner may be a Python class
+that inherits from :class:`avocado.core.nrunner.BaseRunner`, and
+implements at least a matching constructor method, and a ``run()``
+method that should yield dictionary(ies) as result(s).
+
+At a different level, a runner can take the form of an executable that
+follows the ``avocado-runner-$KIND`` naming pattern and conforms to a
+given interface/behavior, including accepting standardized command
+line arguments and producing standardized output.
```

Comment:
Here I agree with @beraldoleal that those lines are related to the development guide.

Reply:
Yep, we'll split that out, hopefully soon!